demo-site/servers/nodejs/app.js

```diff
@@ -0,0 +1,202 @@
+const path = require('path');
+const axios = require('axios');
+const express = require('express');
+const cookieParser = require('cookie-parser');
+const formData = require("express-form-data");
+const perimeterx = require('perimeterx-node-express');
+
+const pxConfigJson = require('./px_config.json');
+
+const PORT = 3000;
+
+var PxMiddleware;
+var PxCdMiddleware;
+let pxConfig;
+var PxCdInterval;
+var pxInstance;
+
+const main = () => {
+    pxConfig = initializeConfigs();
+
+    const app = initializeApp();
+    if (process.env.ENABLE_TEST_ENDPOINTS === "true") {
+        setAdditionalActivityHandler(pxConfig);
+        setCustomParam(pxConfig);
+    }
+    setPxMiddleware(app);
+    setRoutes(app);
+    setTestEndpoints(app);
+    setStaticRoutes(app);
+
+    const server = app.listen(PORT, '0.0.0.0', function () {
+        console.log(`NodeJS sample site is listening on port ${PORT}!`)
+    });
+
+    process.on('SIGINT', () => {
+        console.log('Closing http server...');
+        server.close();
+        process.exit(0);
+    });
+}
+
+const initializeApp = () => {
+    const app = express();
+    app.use(cookieParser());
+    app.use(express.json()); 
+    app.use(express.urlencoded());
+    // support form-data/multipart bodies
+    app.use(formData.parse());
+    app.use(formData.format());
+    app.use(formData.stream());
+    app.use(formData.union());
+    app.use((req, res, next) => {
+        console.log(req.method, req.path);
+        next();
+    })
+    return app;
+}
+
+const setPxMiddleware = (app) => {
+    pxInstance = perimeterx.new(pxConfig)
+    PxMiddleware = pxInstance.middleware;
+    app.use(PxMiddlewareWrap);
+
+    if (pxInstance.cdEnforcer) {
+        PxCdMiddleware = pxInstance.cdMiddleware;
+        PxCdInterval = pxInstance.cdEnforcer.setIntervalId;
+        app.use(PxCdMiddleware);
+    }
+    app.use((req, res, next) => {
+        for (const [name, value] of Object.entries(req.headers)) {
+            res.setHeader(name, value);
+        }
```

Comment:
This was added so that the `px-compromised-credentials` header on the request would be transferred over to the response. It's not enough to transfer just that header, though, because it could change based on the enforcer configuration... but transferring ALL the headers to the response could be problematic? Ideally we would transfer only the `px_compromised_credentials_header` config value.

Reply:
@DanBezalelpx I would modify this to be only the px-compromised-credentials header.